src/helpers/merge-objects/index.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  cloneDeep,
+  isArray,
+  isEqual,
+  isObjectLike,
+  mergeWith,
+} from "lodash";
+
+import { ObjectMap } from "../../types";
+
+export type MergeObjectsMatcher = (key: string, value: any) => any;
+
+export default function mergeObjects(obj: ObjectMap, src: ObjectMap, matcher: MergeObjectsMatcher): ObjectMap {
+  function mergeCustomizer(objValue: any, srcValue: any, key: string): any[] | undefined {
+    if (!isArray(objValue) || !isArray(srcValue)) return undefined;
+    const objValues = objValue as any[];

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)